mesh/mesh.go

```diff
@@ -653,14 +653,7 @@ func (msh *Mesh) SetZeroBlockLayer(lyr types.LayerID) error {
 func (msh *Mesh) AddTXsFromProposal(ctx context.Context, layerID types.LayerID, proposalID types.ProposalID, txIDs []types.TransactionID) error {
 	logger := msh.WithContext(ctx).WithFields(layerID, proposalID, log.Int("num_txs", len(txIDs)))
 	logger.Debug("adding proposal txs to mesh")
-
 	msh.addTransactionsForBlock(logger, layerID, types.EmptyBlockID, txIDs)
-
-	if err := msh.storeTransactionsFromPool(layerID, types.EmptyBlockID, txIDs); err != nil {
```

Comment:
why remove this? this should be the first chance of a TX making into the database.

Reply:
because this code is redundant, please check in addTransactionsForBlock

Reply:
we did this twice, yes. once when adding proposals and once when block is generated/synced. but we want those TXs in DB and the latest layer (most recent gossiped data) to be set as early as possible. i think it should stay as is.

Reply:
@countvonzero please take a look into the code. this code that i removed was is redundant specifically in the code path for proposals

Reply:
you can see that transactions are saved on the line above removed line

Reply:
oh! what?! i must have forgot to delete these 3 lines after creating `addTransactionsForBlock()`. sorry. my bad